dftimewolf/lib/processors/turbinia.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+"""Processes cloud artifacts using a remote Turbinia instance."""
+from __future__ import unicode_literals
+from __future__ import absolute_import
+
+import os
+import tempfile
+
+from dftimewolf.lib.module import BaseModule
+
+from turbinia import client as turbinia_client
+from turbinia import config
+from turbinia import evidence
+from turbinia import output_manager
+from turbinia import TurbiniaException
+from turbinia.message import TurbiniaRequest
+
+
+class TurbiniaProcessor(BaseModule):
+  """Process cloud disks with remote Turbinia instance.
+
+  Attributes:
+    client: A TurbiniaClient object
+    disk_name: Name of the disk to process
+    instance (string): The name of the Turbinia instance
+    project: The project containing the disk to process
+    region (string): The region Turbinia is in
+    zone: The zone containing the disk to process
+    _output_path: The path to output files
+  """
+
+  def __init__(self, state):
+    """Initialize the Turbinia artifact processor object.
+
+    Args:
+      state: The dfTimewolf state object
+    """
+    super(TurbiniaProcessor, self).__init__(state)
+    self.client = None
+    self.disk_name = None
+    self.instance = None
+    self.project = None
+    self.region = None
+    self.zone = None
+    self._output_path = None
+
+  def setup(self, disk_name, project, zone):  # pylint: disable=arguments-differ
+    """Sets up the object attributes.
+
+    Args:
+      disk_name: Name of the disk to process
+      project: The project containing the disk to process
+      zone: The zone containing the disk to process
+    """
+    if self.state.input and not disk_name:
+      _, disk = self.state.input[0]
+      disk_name = disk.name
+      print('Using disk {0:s} from previous collector'.format(disk_name))
+
+    if disk_name is None or project is None or zone is None:
+      self.state.add_error(
+          'disk_name, project or zone are not all specified, bailing out',
+          critical=True)
+      return
+    self.disk_name = disk_name
+    self.project = project
+    self.zone = zone
+    self._output_path = tempfile.mkdtemp()
+
+    try:
+      config.LoadConfig()
```

Comment:
I would import this as `turbinia_config` since we already have a `config` module in dftimewolf and it may be confusing.

Reply:
Done.